### PR TITLE
Capstone 5.0

### PR DESCRIPTION
**Pull request by motorto**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-{glibc,musl}
  - aarch64-musl: cross

- makes one qemu test fail
- Boomerang Regressions: https://github.com/BoomerangDecompiler/boomerang/issues/277

### CODE REVIEW

File: srcpkgs/boomerang/template
Comment (classabbyamp) [inline srcpkgs/boomerang/template:1]: needs python3 in hostmakedepends (and can probably be removed from checkdepends?)

**Comment by classabbyamp** [comment]
would you mind adding this patch to the PR? I've tested it and it works.

```patch
From 681f1c71fe0ffe780407365ebbd91efdfc3fb8b9 Mon Sep 17 00:00:00 2001
From: classabbyamp <void@placeviolette.net>
Date: Tue, 11 Jul 2023 20:19:36 -0400
Subject: [PATCH] imhex: use system capstone

---
 srcpkgs/imhex/template | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/srcpkgs/imhex/template b/srcpkgs/imhex/template
index 3f7f475038..94a058c2b1 100644
--- a/srcpkgs/imhex/template
+++ b/srcpkgs/imhex/template
@@ -1,18 +1,17 @@
 # Template file for 'imhex'
 pkgname=imhex
 version=1.30.1
-revision=1
+revision=2
 build_wrksrc="ImHex"
 build_style=cmake
 build_helper=qemu
-# XXX: when capstone v5 is out, -DUSE_SYSTEM_CAPSTONE=ON
 configure_args="-DIMHEX_OFFLINE_BUILD=ON -DIMHEX_STRIP_RELEASE=OFF
- -DUSE_SYSTEM_CURL=ON -DUSE_SYSTEM_FMT=ON -DUSE_SYSTEM_LLVM=ON
+ -DUSE_SYSTEM_CURL=ON -DUSE_SYSTEM_FMT=ON -DUSE_SYSTEM_LLVM=ON -DUSE_SYSTEM_CAPSTONE=ON
  -DUSE_SYSTEM_YARA=ON -DUSE_SYSTEM_NLOHMANN_JSON=ON -DIMHEX_DISABLE_UPDATE_CHECK=ON"
 hostmakedepends="pkg-config clang-tools-extra"
 makedepends="libcurl-devel fmt-devel llvm jansson-devel yara-devel json-c++
  freetype-devel glfw-devel gtk+3-devel python3-devel file-devel mbedtls-devel
- clang-tools-extra"
+ clang-tools-extra capstone-devel"
 short_desc="Hex editor for reverse engineers and programmers"
 maintainer="classabbyamp <void@placeviolette.net>"
 license="GPL-2.0-or-later"
-- 
2.41.0
```

**Comment by XVilka** [comment]
@motorto I see you updated Rizin to use capstone. It is not currently supported for all releases; it requires https://github.com/rizinorg/rizin/pull/3653 to be merged first to fix all tests and mitigate some capstone regressions.  We will support capstone starting the next release (0.6.0), which is planned soon (a few weeks or so). While it *might* compile, the results could sometimes be incorrect. It's better to wait for the next release.

**Comment by motorto** [comment]
> @motorto I see you updated Rizin to use capstone. It is not currently supported for all releases; it requires [rizinorg/rizin#3653](https://github.com/rizinorg/rizin/pull/3653) to be merged first to fix all tests and mitigate some capstone regressions. We will support capstone starting the next release (0.6.0), which is planned soon (a few weeks or so). While it _might_ compile, the results could sometimes be incorrect. It's better to wait for the next release.

Appreciate your insight. We will wait for the rizin release then.

**Comment by XVilka** [comment]
@motorto [Rizin 0.6.0](https://github.com/rizinorg/rizin/releases/tag/v0.6.0) and Cutter 2.3.0 that support capstone 5.0 are out, sorry for the delay.

**Comment by XVilka** [comment]
@motorto rizin bugfix release is out, fixing [CVE-2023-40022](https://github.com/rizinorg/rizin/security/advisories/GHSA-92h6-wwc2-53cq): https://github.com/rizinorg/rizin/releases/tag/v0.6.1

Corresponding Cutter release was also tagged with updated translations: https://github.com/rizinorg/cutter/releases/tag/v2.3.1

**Comment by motorto** [comment]
> @motorto rizin bugfix release is out, fixing [CVE-2023-40022](https://github.com/rizinorg/rizin/security/advisories/GHSA-92h6-wwc2-53cq): https://github.com/rizinorg/rizin/releases/tag/v0.6.1
> 
> Corresponding Cutter release was also tagged with updated translations: https://github.com/rizinorg/cutter/releases/tag/v2.3.1

Thanks !!!

**Comment by motorto** [comment]
@classabbyamp: added `checkdepends=cmake` on the `imhex` template. It couldn't find the `ctest` without it ...

**Comment by classabbyamp** [comment]
hm that should probably be reported in #45679

**Comment by sgn** [comment]
@motorto  Did you rebase on latest void-packages?

**Comment by XVilka** [comment]
By the way, there is a bugfix release for capstone, you might want to try switching to it directly: https://github.com/capstone-engine/capstone/releases/tag/5.0.1

**Comment by motorto** [comment]
> @motorto Did you rebase on latest void-packages?

Not the latest/latest I would say last week. But will rebase now, make a rebuild and push. Asking because of the cmake build style changes, correct ?

**Comment by sgn** [comment]
> > @motorto Did you rebase on latest void-packages?
> 
> Not the latest/latest I would say last week. But will rebase now, make a rebuild and push. Asking because of the cmake build style changes, correct ?

Yes, the check run normally on top of current master.

**Comment by motorto** [comment]
> Yes, the check run normally on top of current master.

Yes, can confirm that, removed the `checkdepends=` line. Thanks for the attention.

**Comment by XVilka** [comment]
Meanwhile we updated Cutter and Rizin with new bugfix releases, you could pick them in the same PR:
- https://github.com/rizinorg/rizin/releases/tag/v0.6.2
- https://github.com/rizinorg/cutter/releases/tag/v2.3.2

**Pull request merged**